### DISCUSSION TITLE
Rails version attempting to bump from 6.1.6.1 to 6.1.0

### DISCUSSION THREAD

**Discussion by narwold** [renovatebot/renovate#18673] [Request Help] [answered]
Why would Renovate suggest this "upgrade"?

![Screen Shot 2022-10-31 at 9 21 23 AM](https://user-images.githubusercontent.com/1108152/199017597-43515b3e-f860-43aa-89ff-5b828e53ad3d.png)

**Comment by rarkins** [comment, marked as answer]
Looks like a bug, probably because it doesn't understand four part versioning well enough. What does the `packageFiles with updates` log entry say about this? e.g. currentVersion and newVersion

**Reply by narwold** [reply]
I tracked it down in code. It appears it's part of `lib/modules/versioning/ruby/strategies/replace.ts`. When the current version is 6.1.6.1 and it tries to replace based on a target version of 6.1.7, it does a "replace" on the 6.1.7, yielding 6.1.0. I'll have a testless PR up shortly to at least illustrate the problem and perhaps provide a potential solution (early form).

**Reply by narwold** [reply]
PR: https://github.com/renovatebot/renovate/pull/18679

**Reply by narwold** [reply]
This can be closed
